### PR TITLE
Auto detect OS and get appropriate kubectl

### PR DESCRIPTION
**Pull request by lwr20**

### CODE REVIEW

File: gce/etcd-config-01.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    permissions: 0644
+    content: |
+      127.0.0.1    localhost
+
+coreos:
+  update:
+    reboot-strategy: off
+  etcd2:
+    name: etcd-00
+    initial-cluster: etcd-00=http://kube-scale-etcd-01:2380,etcd-01=http://kube-scale-etcd-02:2380,etcd-02=http://kube-scale-etcd-03:2380
```
Comment (caseydavenport) [inline gce/etcd-config-01.yaml]: Is there any difference between `etcd-config-01`, `etcd-config-02`, etc?

They look the same to me, but it's hard to tell.

**Comment by lwr20** [comment]
```
$ diff etcd-config-01.yaml etcd-config-02.yaml 
17c17
<     name: etcd-00
---
>     name: etcd-01
```
Just the host's name.  

I'm certain there's a more elegant way to take a template file and substitute variables into it, but it wasn't obvious to me how to do that in a Makefile.  If the makefile was a python script, it'd be easy ;)

**Comment by caseydavenport** [comment]
@lwr20 Feel free to re-write the Makefile as Python if you'd like :)

Or, just a subset of it.  I think the goal is to expand this to use a 5-node cluster anyway once this has been tested (which I think it has now).

File: gce/Makefile [hunk 2]
```diff
@@ -83,6 +89,32 @@ gce-create: kubectl calicoctl
 	make --no-print-directory gce-forward-ports
 	#make --no-print-directory apply-node-labels
 
+gce-create-etcd:
+	gcloud compute instances create \
```
Comment (caseydavenport) [inline gce/Makefile]: This is going to spin up each server sequentially.  You can make this quicker by sticking an & at the end of the first two calls and the nodes will be started in parallel.  I think that is what we want in order to minimize cluster startup time.

One of the goals here is to be able to start up a cluster as fast as possible.

Comment (lwr20) [inline gce/Makefile]: Done.

**Comment by caseydavenport** [comment]
LGTM

**Pull request merged**